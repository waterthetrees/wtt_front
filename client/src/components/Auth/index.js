export { Loading } from './Loading';
export { AuthButton, ProfileAuthButton } from './AuthButton';
export { RedirectWithHash } from './RedirectWithHash';
export { RequireAuth } from './RequireAuth';
export { Auth0ProviderWithRedirect } from './Auth0ProviderWithRedirect';
