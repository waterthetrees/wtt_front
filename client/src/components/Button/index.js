export * from './Button';
export * from './ButtonShowHide';
