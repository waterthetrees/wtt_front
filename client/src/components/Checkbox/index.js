export { default as AdoptionCheckbox } from './AdoptionCheckbox';
export { default as InfoCheckbox } from './InfoCheckbox';
export { default as StarCheckbox } from './StarCheckbox';
