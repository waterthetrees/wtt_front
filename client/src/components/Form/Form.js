import React from 'react';
import { FormProvider } from 'react-hook-form';

export default function Form({
  children, onSubmit, ...methods
}) {
  return (
    <FormProvider {...methods}>
      <form
        onSubmit={onSubmit}
      >
        {children}

        {/* Add a hidden input to the form so that automatic submission when pressing enter is
          handled by the browser. */}
        <input type="submit" style={{ display: 'none' }} />
      </form>
    </FormProvider>
  );
}
