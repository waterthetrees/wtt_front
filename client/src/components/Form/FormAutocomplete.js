/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import {
  TextField,
  Autocomplete,
} from '@mui/material';
import { Controller } from 'react-hook-form';
import FormErrorMessage from './FormErrorMessage';

export default function FormAutocomplete({
  name, label, options = [], rules, ...props
}) {
  return (
    <>
      <Controller
        name={name}
        rules={rules}
        render={({ field }) => (
          <Autocomplete
            {...field}
            options={options}
            freeSolo
            autoSelect
            handleHomeEndKeys
            onChange={(_, value) => field.onChange(value)}
            renderInput={(params) => (
              <TextField
                {...params}
                label={label}
                variant="standard"
              />
            )}
            {...props}
          />
        )}
      />
      <FormErrorMessage name={name} label={label} />
    </>
  );
}
