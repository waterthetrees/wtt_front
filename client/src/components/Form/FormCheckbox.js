import React from 'react';
import { Checkbox, FormControlLabel } from '@mui/material';
import { Controller } from 'react-hook-form';
import FormErrorMessage from './FormErrorMessage';

export default function FormCheckbox({ name, label, rules, sx, ...props }) {
  // We have to wrap the Controller around the Checkbox, not the FormControlLabel, since the
  // Checkbox is what changes the actual value.
  return (
    <>
      <FormControlLabel
        label={label}
        sx={sx}
        control={
          <Controller
            name={name}
            rules={rules}
            render={({ field }) => (
              <Checkbox {...field} checked={field.value} {...props} />
            )}
          />
        }
      />
      <FormErrorMessage name={name} label={label} />
    </>
  );
}
