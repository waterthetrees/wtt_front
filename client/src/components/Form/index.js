export { default as Form } from "./Form";
export { default as FormAutocomplete } from "./FormAutocomplete";
export { default as FormCheckbox } from "./FormCheckbox";
export { default as FormDecimalField } from "./FormDecimalField";
export { default as FormErrorMessage } from "./FormErrorMessage";
export { default as FormRadio } from "./FormRadio";
export { default as FormRadioGroup } from "./FormRadioGroup";
export { default as FormScrollableDialog } from "./FormScrollableDialog";
export { default as FormSelect } from "./FormSelect";
export { default as FormTextField } from "./FormTextField";
export { default as FormTreeGroup } from "./FormTreeGroup";
