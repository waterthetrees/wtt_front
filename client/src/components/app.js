
import React, { useState } from 'react';
import { Switch, Route, Redirect, withRouter } from 'react-router-dom';
import {SWRConfig} from 'swr';

import LayoutComponent from '../components/Layout/Layout';
import NotFoundComponent from '../pages/notFound/NotFound';
// import 'mapbox-gl/dist/mapbox-gl.css';

const PrivateRoute = ({ component, isAuthenticated, email, ...rest }) => (

    <Route
      {...rest}
      render={props =>
        isAuthenticated
        ? React.createElement(component, props)
        : (<Redirect to={{pathname: '/login', state: { from: props.location }}} />)
      }
    />
  );

const App = (props) => {
  const component_name = 'App!!!!\n\n\n';
  // const email = useSelector(state => state.auth.email);
  const email = 'testrose@gamma.com';
  // const isAuthenticated = useSelector(state => state.auth.isAuthenticated);
  const isAuthenticated = true;
  return (
    <SWRConfig 
      value={{
        fetcher: (...args) => fetch(...args).then(res => res.json())
      }}
    >
    <Switch>
      <Route path="/" exact render={() => <Redirect to="/mapper"  />} />
      <PrivateRoute
        isAuthenticated={isAuthenticated}
        email={email}
        path="/"
        component={LayoutComponent}
      />
      <Route component={NotFoundComponent} />
    </Switch>
    </SWRConfig>
  );
}

export default withRouter(App);