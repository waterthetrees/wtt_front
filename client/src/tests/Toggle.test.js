import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Toggle } from '@/components/Form';

describe('<Toggle /> spec', () => {
  beforeEach(() => {
    localStorage.clear();
  });

  it('Renders the Toggle component correctly', () => {
    const mockSetView = jest.fn();
    const { container } = render(<Toggle view="list" setView={mockSetView} />);
    expect(container).toMatchSnapshot();
  });

  it('Changes the view on button click and updates local storage', () => {
    const mockSetView = jest.fn();
    const { getByLabelText } = render(
      <Toggle view="list" setView={mockSetView} />,
    );
    const cardToggleButton = getByLabelText('card');

    fireEvent.click(cardToggleButton);
    expect(mockSetView).toHaveBeenCalledWith('card');
    expect(localStorage.getItem('view')).toEqual('card');
  });

  it('Defaults to the view from props if local storage is not present', () => {
    const mockSetView = jest.fn();
    const { getByTestId } = render(
      <Toggle view="list" setView={mockSetView} />,
    );
    const listToggleButton = getByTestId('list-toggle-button');
    expect(listToggleButton.getAttribute('aria-pressed')).toEqual('true');
  });
});
