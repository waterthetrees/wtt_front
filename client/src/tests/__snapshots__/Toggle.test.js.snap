// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> spec Renders the Toggle component correctly 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 4px;
  background-color: white;
}

.emotion-0 .MuiToggleButtonGroup-grouped:not(:first-of-type) {
  margin-left: -1px;
  border-left: 1px solid transparent;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.emotion-0 .MuiToggleButtonGroup-grouped:not(:last-of-type) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-0 .MuiToggleButtonGroup-grouped.Mui-selected+.MuiToggleButtonGroup-grouped.Mui-selected {
  border-left: 0;
  margin-left: 0;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  border-radius: 4px;
  padding: 11px;
  border: 1px solid rgba(0, 0, 0, 0.12);
  color: rgba(0, 0, 0, 0.54);
}

.emotion-1::-moz-focus-inner {
  border-style: none;
}

.emotion-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-1.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-1:hover {
    background-color: transparent;
  }
}

.emotion-1.Mui-selected {
  color: #2e7d32;
  background-color: rgba(46, 125, 50, 0.08);
}

.emotion-1.Mui-selected:hover {
  background-color: rgba(46, 125, 50, 0.12);
}

@media (hover: none) {
  .emotion-1.Mui-selected:hover {
    background-color: rgba(46, 125, 50, 0.08);
  }
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-3 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div
    aria-label="view"
    class="MuiToggleButtonGroup-root emotion-0"
    role="group"
  >
    <button
      aria-label="column"
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-success MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-1"
      data-testid="column-toggle-button"
      tabindex="0"
      type="button"
      value="column"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
        data-testid="ViewColumnIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M14.67 5v14H9.33V5h5.34zm1 14H21V5h-5.33v14zm-7.34 0V5H3v14h5.33z"
        />
      </svg>
       column
      <span
        class="MuiTouchRipple-root emotion-3"
      />
    </button>
    <button
      aria-label="quilt"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-success MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-1"
      data-testid="quilt-toggle-button"
      tabindex="0"
      type="button"
      value="quilt"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
        data-testid="ViewQuiltIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M21 5v6.5H9.33V5H21zm-6.33 14v-6.5H9.33V19h5.34zm1-6.5V19H21v-6.5h-5.33zM8.33 19V5H3v14h5.33z"
        />
      </svg>
       quilt
      <span
        class="MuiTouchRipple-root emotion-3"
      />
    </button>
  </div>
</div>
`;
